bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -140,6 +160,54 @@ public static PercentType[] hsbToRgbPercent(HSBType hsb) {
         return new PercentType[] { red, green, blue };
     }
 
+    /**
+     * Transform <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType} to RGBW.
+     *
+     * See <a href=
+     * "https://stackoverflow.com/questions/40312216/converting-rgb-to-rgbw">Converting RGB to RGBW</a>.
+     *
+     * This function does not round the components. For conversion to integer values in the range 0 to 255 use
+     * {@link #hsbToRgb(HSBType)}.
+     *
+     * See also: {@link #hsbToRgb(HSBType)}, {@link #hsbTosRgb(HSBType)}, {@link #hsbToRgbPercent(HSBType)}
+     *
+     * @param hsb an {@link HSBType} value.
+     * @return array of four {@link PercentType} with the RGBW values in the range 0 to 100 percent.
+     */
+    public static PercentType[] hsbToRgbwPercent(HSBType hsb) {
+        PercentType[] rgb = hsbToRgbPercent(hsb);
+        final BigDecimal Ri = rgb[0].toBigDecimal();
+        final BigDecimal Gi = rgb[1].toBigDecimal();
+        final BigDecimal Bi = rgb[2].toBigDecimal();
+        // Get the maximum between R, G, and B
+        final BigDecimal tM = Ri.max(Gi.max(Bi));
+
+        // If the maximum value is 0, immediately return pure black.
+        if (tM.floatValue() == 0) {
+            return new PercentType[] { PercentType.ZERO, PercentType.ZERO, PercentType.ZERO, PercentType.ZERO };
+        }
+
+        // This section serves to figure out what the color with 100% hue is
+        final BigDecimal multiplier = BIG_DECIMAL_100.divide(tM, 0, RoundingMode.DOWN);
+        final BigDecimal hR = Ri.multiply(multiplier);
+        final BigDecimal hG = Gi.multiply(multiplier);
+        final BigDecimal hB = Bi.multiply(multiplier);
+
+        // This calculates the Whiteness (not strictly speaking Luminance) of the color
+        final BigDecimal M = hR.max(hG.max(hB));
+        final BigDecimal m = hR.min(hG.min(hB));
+        final BigDecimal Luminance = ((M.add(m).divide(BIG_DECIMAL_2).subtract(BIG_DECIMAL_50))
+                .multiply(BIG_DECIMAL_100.divide(BIG_DECIMAL_50))).divide(multiplier);
+
+        // check range
+        final BigDecimal Ro = Ri.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Go = Gi.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Bo = Bi.subtract(Luminance).max(BigDecimal.ZERO);
+        final BigDecimal Wo = Luminance.max(BigDecimal.ZERO);
```

Comment:
```suggestion
        BigDecimal outRed = Ri.subtract(Luminance).max(BigDecimal.ZERO);
        BigDecimal outGreen = Gi.subtract(Luminance).max(BigDecimal.ZERO);
        BigDecimal outBlue = Bi.subtract(Luminance).max(BigDecimal.ZERO);
        BigDecimal outWhite = Luminance.max(BigDecimal.ZERO);
```

no need to add final to local variables,